### PR TITLE
Autopilot for timetable mode

### PR DESCRIPTION
**Pull request by Csantucci**
See https://blueprints.launchpad.net/or/+spec/autopilot-for-timetable
and https://www.elvastower.com/forums/index.php?/topic/38134-autopilot-in-timetable-mode-unstable/

This PR covers strictly only autopilot. The other timetable features described in the ET thread are not included and might be part of later PRs.

### CODE REVIEW

File: Source/Orts.Simulation/Simulation/Timetables/TTTrain.cs [hunk 1]
```diff
@@ -13298,6 +13520,7 @@ public bool PerformDetach(TTTrain train, bool allowPlayerSelect)
                 Trace.TraceInformation("Train {0} : detach to train {1} : cannot find new train \n", train.Name, DetachFormedTrainName);
             }
 
+//            newTrain.Autopilot = train.Autopilot;
```
Comment (SteelFill) [inline Source/Orts.Simulation/Simulation/Timetables/TTTrain.cs:13523]: I see there are a couple of lines of functional code that are commented out, if you don't see any use for these lines anymore they might as well be removed to improve readability.

**Review by SteelFill — COMMENTED** [review]
Seems like a very handy feature. I've let the AI autopilot loose on some simpler timetables (I don't think I have any downloaded that include more of the complicated features) without any crashes or behavior that's unusual for autopilot.

In the meantime I scanned over the code and pointed out a set of style things, though I don't have much insight for the actual functionality.

File: Source/Orts.Simulation/Simulation/Simulator.cs [hunk 2]
```diff
@@ -1600,7 +1601,7 @@ private void RestoreTrains(BinaryReader inf)
             // find player train
             foreach (Train thisTrain in Trains)
             {
-                if (thisTrain.TrainType == Train.TRAINTYPE.PLAYER
+                if (thisTrain.TrainType == Train.TRAINTYPE.PLAYER || thisTrain is TTTrain && thisTrain == Trains[0]
```
[the remark this answers is not in the document]
Comment (Csantucci) [reply Source/Orts.Simulation/Simulation/Simulator.cs]: Done

File: Source/Orts.Simulation/Simulation/Timetables/TTTrain.cs [hunk 3]
```diff
@@ -2384,7 +2388,7 @@ public bool StartFromAITrain(TTTrain otherTrain, int presentTime, TrackCircuitSe
                 LeadLocomotiveIndex = otherTrain.LeadLocomotiveIndex;
 
                 // Copy other train speed if not restricted for either train
-                if (!otherTrain.SpeedSettings.restrictedSet && !SpeedSettings.restrictedSet)
+                if (!otherTrain.SpeedSettings.restrictedSet && !SpeedSettings.restrictedSet || TrainMaxSpeedMpS == 0)
```
[the remark this answers is not in the document]
Comment (Csantucci) [reply Source/Orts.Simulation/Simulation/Timetables/TTTrain.cs]: Done

File: Source/Orts.Simulation/Simulation/Timetables/TTTrain.cs [hunk 4]
```diff
@@ -3964,14 +4046,30 @@ public override void UpdateStationState(float elapsedClockSeconds, int presentTi
...
 
             ResetActions(true);
         }
 
+        //================================================================================================//
+        /// <summary>
+        /// If autopiloted, generate strings for NextStationWindow
+        /// Override for AITrain class
+        /// <\summary>
+        public void PopulateNextStationWindow()
```
[the remark this answers is not in the document]
Comment (Csantucci) [reply Source/Orts.Simulation/Simulation/Timetables/TTTrain.cs]: Done

File: Source/Orts.Simulation/Simulation/Timetables/TTTrain.cs (hunk 1 above)
Comment (Csantucci) [reply Source/Orts.Simulation/Simulation/Timetables/TTTrain.cs]: Done

**Review by cjakeman — APPROVED** [review]
Hi Carlo,

Pleased to approve this. I'll leave you to merge it yourself.

Chris

**Pull request merged**